microcosm_flask/forwarding.py

```diff
@@ -19,21 +19,36 @@ def use_forwarded_port(graph):
     The URL adapter is used by `url_for` to build a URLs.
 
     """
-    forwarded_port = request.headers.get("X-Forwarded-Port")
-    if not forwarded_port:
-        return None
+    # There must be a better way!
+    context = _request_ctx_stack.top
 
     if _request_ctx_stack is None:
         return None
 
-    # There must be a better way!
-    context = _request_ctx_stack.top
-    context.url_adapter.server_name = "{}:{}".format(
-        context.url_adapter.server_name.split(":", 1)[0],
-        forwarded_port,
-    )
+    # determine the configured overrides
+    forwarded_host = graph.config.port_forwarding.get("host")
```

Comment:
I piggy-backed on existing "port forwarding" functionality. It's a poor naming, but it's already used everywhere.